analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Conversion.cs

```diff
@@ -18,18 +18,40 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
+using SonarAnalyzer.SymbolicExecution.Constraints;
+
 namespace SonarAnalyzer.SymbolicExecution.Roslyn.OperationProcessors;
 
-internal sealed class Conversion : SimpleProcessor<IConversionOperationWrapper>
+internal sealed class Conversion : MultiProcessor<IConversionOperationWrapper>
 {
     protected override IConversionOperationWrapper Convert(IOperation operation) =>
         operation.ToConversion();
 
-    protected override ProgramState Process(SymbolicContext context, IConversionOperationWrapper conversion) =>
-        context.State[conversion.Operand] is { } value && IsBuildIn(conversion.OperatorMethod) // Built-in conversions only
-            ? context.SetOperationValue(value)
-            : context.State;
+    protected override ProgramState[] Process(SymbolicContext context, IConversionOperationWrapper conversion) =>
+        IsNotBuildIn(conversion.OperatorMethod) ? context.State.ToArray() : StateForBuildInConversion(context, conversion);
+
+    private static bool IsNotBuildIn(ISymbol symbol) =>
```

Comment:
Why is this negative? Negative is always one more headache. Conditions should be positive when possible

Reply:
Because it made the implementation of the method simpler. Do we prioritize simple names or simple implementation? I'll swap it for now.

Reply:
Simple names with simple implementations :D 
That's a delicate balance that doesn't have a clear answer.

Reply:
In this case, I would say the positive name is better. And the implementation was clear and easy to follow (not based on ugly negations)